org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/JavaSearchScopeTests.java

```diff
@@ -1124,4 +1141,68 @@ public void testBug397818() throws CoreException {
+ * https://github.com/eclipse-jdt/eclipse.jdt.core/issues/332
+ */
+public void testModuleJarSearchBugGh332() throws Exception {
+	String testProjectName = "gh322ModuleJarSearchBug";
+	try {
+		IJavaProject project = setUpJavaProject(testProjectName, "11", false);
+		waitForAutoBuild();
+		waitUntilIndexesReady();
+
+		JavaSearchResultCollector resultCollector = new JavaSearchResultCollector();
+		String timerTypeFqn = "java.util.Timer";
+		IType timerType = project.findType(timerTypeFqn);
+		assertNotNull("Failed to find JRE type: " + timerTypeFqn, timerType);
+		String methodName = "sched";
+		IMethod schedMethod = null;
+		for (IMethod method : timerType.getMethods()) {
+			if (methodName.equals(method.getElementName())) {
+				schedMethod = method;
+			}
+		}
+		IJavaSearchScope scope = SearchEngine.createJavaSearchScope(new IJavaElement[] { schedMethod.getAncestor(IJavaElement.PACKAGE_FRAGMENT) });
+		assertNotNull("Failed to find method: " + methodName + ", in JRE type: " + timerTypeFqn, schedMethod);
+		int agnosticMatchRule = SearchPattern.R_EXACT_MATCH | SearchPattern.R_CASE_SENSITIVE | SearchPattern.R_ERASURE_MATCH;
+		SearchPattern pattern = SearchPattern.createPattern(schedMethod, IJavaSearchConstants.REFERENCES, agnosticMatchRule);
+		SearchEngine searchEngine = new SearchEngine();
+		searchEngine.search(pattern, new SearchParticipant[] {SearchEngine.getDefaultSearchParticipant()}, scope, resultCollector, new NullProgressMonitor());
+
+		String foundReferences = resultCollector.toString();
+		assertFalse("Expected search to find references of method: " + schedMethod, foundReferences.isEmpty());
+		List<String> results = Arrays.asList(foundReferences.split(System.lineSeparator()));
+		String[] expectedResults = {
+				"void java.util.Timer.schedule(java.util.TimerTask, long)",
+				"void java.util.Timer.schedule(java.util.TimerTask, java.util.Date)",
+				"void java.util.Timer.schedule(java.util.TimerTask, long, long)",
+				"void java.util.Timer.schedule(java.util.TimerTask, java.util.Date, long)",
+				"void java.util.Timer.scheduleAtFixedRate(java.util.TimerTask, long, long)",
+				"void java.util.Timer.scheduleAtFixedRate(java.util.TimerTask, java.util.Date, long)",
+		};
+		assertEquals("Unexpected search results:\n" + foundReferences, expectedResults.length, results.size());
+		Set<String> notFound = new LinkedHashSet<>();
+		notFound.addAll(Arrays.asList(expectedResults));
+		for (String expectedResult : expectedResults) {
+			boolean isFound = false;
+			for (String match : results) {
+				if (match.contains(expectedResult)) {
+					isFound = true;
+					break;
+				}
+			}
+			if (isFound) {
+				notFound.remove(expectedResult);
+			}
+		}
+		assertTrue("Unexpected search result\nExpected:\n" + String.join("\n", expectedResults) + "\nActual:\n" + foundReferences + "\nExpected but not found:\n" + String.join("\n", notFound),
+				notFound.isEmpty());
+	} finally {
+		JavaCore.setOptions(getDefaultJavaCoreOptions());
```

Comment:
> Why is JavaCore.setOptions call needed? I don't see it above

The test is changing some options and fails on tear down because of that. I don't know which part of the test changes the options though. I can debug to find out if you want, to me it was simpler to reset the options.